src/include/duckdb/common/arrow/appender/append_data.hpp

```diff
@@ -47,13 +47,15 @@ struct ArrowAppendData {
 
 	// the arrow array C API data, only set after Finalize
 	unique_ptr<ArrowArray> array;
-	duckdb::array<const void *, 3> buffers = {{nullptr, nullptr, nullptr}};
+	duckdb::array<const void *, 4> buffers = {{nullptr, nullptr, nullptr, nullptr}};
```

Comment:
Just a note, we keep the ArrowAppendData alive, it serves as the `private_data` for our created arrow arrays, this now always creates 4 buffers
Probably not an issue though